279 - enable to write a second value converter for the same field

winklertomas opened the pull request:
### Motivation

Fixes #279. Enables to write a second value converter for the same field and adds a way to map to Kontent content item field without interfering with Newtonsoft JSON serializer. Can be used for example for linked item codenames as is demonstrated in this PR

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

Simply007 reviewed (changes requested):
* rename `FieldNameAttribute` to `PropertyNameAttribute` (describe in XMLdoc)

Q: -> extendability of FieldName -> consequences, Current possibilities of JsonProperty
A: It is still possible to use JsonProperty

TODO review:
* distributed caching and binary serialization => check if we have tests for https://github.com/Kentico/kontent-delivery-sdk-net/wiki/Caching-responses#distributedcachemanager -> extend https://github.com/Kentico/kontent-delivery-sdk-net/blob/master/Kentico.Kontent.Delivery.Caching.Tests/TestItem.cs tie one property with custom attribute
* Describe the attribute: https://github.com/Kentico/kontent-delivery-sdk-net/wiki/Working-with-strongly-typed-models#naming-the-properties

winklertomas commented:
I've renamed the FieldNameAttribute to PropertyNameAttribute and added the attribute to the property of TestItem.cs to show that it doesn't affect caching... 

In my opinion, the wiki should be updated after the PR is merged to the master branch and it should look e.g. as follows:

If you need to change the codename of an element that a property is bound to, you can enrich the property with the Kentico.Kontent.Delivery.ContentItems.PropertyNameAttribute attribute.

[PropertyName("text_field")]
public string ArticleTitle { get; set; }

If you also want to set the behavior of the Newtonsoft and don't want to map the same element to different properties, you can use the Newtonsoft.Json.JsonPropertyAttribute.

[JsonProperty("text_field")]
public string ArticleTitle { get; set; }

MiroKentico commented on Kentico.Kontent.Delivery/ContentItems/PropertyNameAttribute.cs:
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Kentico.Kontent.Delivery.ContentItems
```
What about move of this whole class to namespace (and appropriate folder) `Kentico.Kontent.Delivery.ContentItems.Attributes` ? The benefit would be better structure.

I am thinking about this, because for instance some custom attributes for RichText are already in separate namespace `Kentico.Kontent.Delivery.ContentItems.RichText.Attributes` (and not in root of `...RichText`).

MiroKentico replied:
Thanks for changing this.

MiroKentico reviewed (changes requested):
There are few comments, please check them.

MiroKentico commented on Kentico.Kontent.Delivery/ContentItems/PropertyMapper.cs:
```diff
@@ -26,6 +27,13 @@ public bool IsMatch(PropertyInfo modelProperty, string fieldName, string content
                 return false;
             }
 
+            PropertyNameAttribute fieldNameAttr = modelProperty.GetCustomAttribute<PropertyNameAttribute>();
```
Thanks @Sevitas for implementation,

MiroKentico reviewed (approved):
Comments were applied, approve.
